### ISSUE TITLE
SIAB allows Control-W to pass onto browser and closes window

### ISSUE THREAD

**Issue by KLuka** [shellinabox/shellinabox#153]
_From [lpsan...@gmail.com](https://code.google.com/u/116498127598887791078/) on December 01, 2011 06:51:01_

What steps will reproduce the problem? 1. Login to Linux system
2. Start up pico or nano text editor
3. Hit control-w on keyboard start a file search What is the expected output? What do you see instead? Expect to see search query bar at bottom of screen.  Instead, the window or tab is closed. What version of the product are you using? On what operating system? Google Chrome 16.0.912.44
Platform 1193.65.0 (Official Build) beta-channel x86-mario
Firmware Mario.03.60.1120.0038G5.0018d

Also applies to Chrome browser v15.x.x.x on OSX 10.6.7. Please provide any additional information below.

_Original issue: http://code.google.com/p/shellinabox/issues/detail?id=153_

**Comment by KLuka** [comment]
_From [lpsan...@gmail.com](https://code.google.com/u/116498127598887791078/) on November 30, 2011 21:52:20_

This can probably be fixed in the keyboard/window event handlers.  Investigating.

**Comment by KLuka** [comment]
_From [lpsan...@gmail.com](https://code.google.com/u/116498127598887791078/) on January 17, 2013 11:05:37_

Discovered fix a while back.  Need to capture & discard the window.onbeforeunload event.

**Comment by KLuka** [comment]
This was fixed with commit 38362b591e64902d8071efcd9fe7c65a19e5501e for browsers other than Chrome. In Chrome user gets "Window Close" warning but Ctrl-W event is not sent to terminal.

Closing this for now.

**Comment by a-detiste** [comment]
> 2. Start up pico or nano text editor

nano can be instructed to use other key-bindings, thhis could be added to the `BUGS`of the man-page.

(W=Whereis, S=Search, F=Find)

```
pi@raspberrypi ~ $ cat .nanorc
bind ^S whereis main
bind ^F whereis main
```
